align_system/utils/alignment_utils.py

```diff
@@ -91,16 +112,21 @@ def __call__(self, kdma_values, target_kdmas, misaligned=False, kde_norm='global
         _check_if_targets_are_kde(target_kdmas)
 
         # Sample KDEs to get scalar targets
-        sampled_target_kdmas = []
+        self.sampled_target_kdmas = []
         for target_kdma in target_kdmas:
             sampled_target_kdma = {'kdma':target_kdma.kdma}
             target_kde = kde_utils.load_kde(target_kdma, kde_norm)
             sampled_target_kdma['value']= float(target_kde.sample(1)) # sample returns array
-            sampled_target_kdmas.append(sampled_target_kdma)
+            self.sampled_target_kdmas.append(sampled_target_kdma)
 
         # Use avergae distance to sampled scalar targets
         AlignmentFunc = AvgDistScalarAlignment()
-        return AlignmentFunc(kdma_values, sampled_target_kdmas, misaligned)
+        return AlignmentFunc(kdma_values, self.sampled_target_kdmas, misaligned=misaligned)
+    
+    def get_best_sample_index(self, kdma_values, target_kdmas, selected_choice, misaligned=False, kde_norm=None):
+        # Use avergae distance to sampled scalar targets
+        AlignmentFunc = AvgDistScalarAlignment()
```

Comment:
I think lowercase underscore separated is more standard for variables (I can maybe understand using it for `AlignmentFunc = AvgDistScalarAlignment` but since you're instantiating `AvgDistScalarAlignment` with `()`, it's an object now not a class)

Reply:
Fixed here https://github.com/ITM-Kitware/align-system/pull/98/commits/e88a7524b644ebdadb0448ef5aa4fadf7d67a8c3